create_k3s_sysext.sh

```diff
@@ -29,5 +29,71 @@ rm -rf "${SYSEXTNAME}"
 mkdir -p "${SYSEXTNAME}"/usr/local/bin
 curl -o "${SYSEXTNAME}/usr/local/bin/k3s" -fsSL "${URL}"
 chmod +x "${SYSEXTNAME}"/usr/local/bin/k3s
+
+mkdir -p "${SYSEXTNAME}"/usr/local/lib/systemd/system/
+cat > "${SYSEXTNAME}"/usr/local/lib/systemd/system/k3s.service << EOF
+[Unit]
+Description=Lightweight Kubernetes
+Documentation=https://k3s.io
+Wants=network-online.target
+After=network-online.target
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
```suggestion
UpheldBy=multi-user.target
```
This works with systemd 255 when you pre-create a symlink with `.upholds/`. Otherwise you can keep `Wantedby` for systemd 252 and create a multi-user-target drop in that sets `Upholds=k3s.service`.

Same for the agent service.

Reply:
For reference, see `create_kubernetes_sysext.sh`

Reply:
don't we ship "systemctl restart multi-user.target` in `ensure-sysext.service` so that users don't have to complicate their sysexts with "Upholds"/"UpheldBy"?

Reply:
I'll wait for a final answer here, before adapting.

Reply:
> don't we ship `systemctl restart multi-user.target` in `ensure-sysext.service` so that users don't have to complicate their sysexts with "Upholds"/"UpheldBy"?

Yes, but that was a workaround until we have a proper way to do this upstream with `.upholds` symlinks that `systemctl enable` will create for `UpheldBy`. The target restart is sometime we want to get rid of because it might create a race condition. Later on when we would improve the overlay setup to be done earlier the `.wants` might work again if the extension is loaded at boot time but not for extensions added/reloaded later. Also, the extensions are build for `OS=_any` and other distros would not ship the Flatcar workaround.

Reply:
Thanks for the clarification, fixed in 289f38e